plugin.go

```diff
@@ -205,6 +206,10 @@ func initGit(branch string) *exec.Cmd {
 	return appendEnv(exec.Command("git", "init", "-b", branch), defaultEnvVars...)
 }
 
+func safeDirectory(safeDirectory string) *exec.Cmd {
+	return appendEnv(exec.Command("git", "config", "safe.directory", os.ExpandEnv(safeDirectory)), defaultEnvVars...)
```

Comment:
That would mean there is an env var in the safeDirectory. However by `EnvVars: []string{"PLUGIN_SAFE_DIRECTORY", "CI_WORKSPACE"},` it should just be the content of those strings. 🤔 

@pat-s Could you log and tell us the content of your `safeDirectory` var?

Reply:
I think this was only needed back when there was `Value:   "$CI_WORKSPACE",` used.

Reply:
Removing `os.ExpandEnv()` works - just tested it. 

What doesn't work in return is the removal of the `--global` flag

```
+ git init -b main
Initialized empty Git repository in /woodpecker/src/[..]
+ git config safe.directory /woodpecker/src/[..]
fatal: not in a git directory
```

Reply:
One reason could be that the `.git` cannot be read as it's owned by another user and hence the call without `--global` fails as this would aim to write into this local `.git` dir.

Adding `--global` puts the setting into a different location, i.e. the one from the user which calls it and still only allows `safe.directory` for the respective directory specified.

I guess we should keep `--global` therefore, unless somebody is able to track this down in more detail?

Reply:
imho we should set safe.directory only if one is not set already otherwise this will break parallel  jobs on local/ssh instances